benchmark/stats/histogram.go

```diff
@@ -123,36 +66,75 @@ func NewHistogram(opts HistogramOptions) *Histogram {
 		opts.BaseBucketSize = 1.0
 	}
 	h := Histogram{
-		opts:         opts,
-		buckets:      make([]bucketInternal, opts.NumBuckets),
-		count:        newCounter(),
-		sum:          newCounter(),
-		sumOfSquares: newCounter(),
-		tracker:      newTracker(),
+		Buckets: make([]HistogramBucket, opts.NumBuckets),
+		Min:     math.MaxInt64,
+		Max:     math.MinInt64,
 
+		opts:                          opts,
 		logBaseBucketSize:             math.Log(opts.BaseBucketSize),
 		oneOverLogOnePlusGrowthFactor: 1 / math.Log(1+opts.GrowthFactor),
 	}
 	m := 1.0 + opts.GrowthFactor
 	delta := opts.BaseBucketSize
-	h.buckets[0].lowBound = float64(opts.MinValue)
-	h.buckets[0].count = newCounter()
+	h.Buckets[0].LowBound = float64(opts.MinValue)
 	for i := 1; i < opts.NumBuckets; i++ {
-		h.buckets[i].lowBound = float64(opts.MinValue) + delta
-		h.buckets[i].count = newCounter()
+		h.Buckets[i].LowBound = float64(opts.MinValue) + delta
 		delta = delta * m
 	}
 	return &h
 }
 
+// Print writes textual output of the histogram values.
+func (h Histogram) Print(w io.Writer) {
```

Comment:
u need to make the receiver consistent. You need to use pointer receiver for consistency.